### PR TITLE
[Spark] Upgrade Spark version to 3.5.2

### PR DESCRIPTION
**Pull request by vkorukanti**
Run CI jobs with Spark 3.5.2 RC2

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -155,6 +155,7 @@ def crossSparkSettings(): Seq[Setting[_]] = getSparkVersion() match {
     targetJvm := "8",
     // For adding staged Spark RC versions, e.g.:
     // resolvers += "Apache Spark 3.5.0 (RC1) Staging" at "https://repository.apache.org/content/repositories/orgapachespark-1444/",
+    resolvers += "Apache Spark 3.5.2 (RC2) Staging" at "https://repository.apache.org/content/repositories/orgapachespark-1458/",
```
Comment (yaooqinn) [inline build.sbt:158]:
```suggestion
    resolvers += "Apache Spark 3.5.2 (RC3) Staging" at "https://repository.apache.org/content/repositories/orgapachespark-1459/",
```

**Comment by yaooqinn** [comment]
Hi @vkorukanti. 
Spark 3.5.2 (RC3) is polling. Please help Spark community to verify it.

Thanks,
Kent Yao

File: .github/workflows/spark_test.yaml [hunk 2]
```diff
@@ -60,7 +60,8 @@ jobs:
           # `-SNAPSHOT` in version (e.g. `3.3.0-SNAPSHOT`) as the version is picked up from
           # the`version.sbt` file.
           pipenv run pip install pip==24.0 setuptools==69.5.1 wheel==0.43.0
-          pipenv run pip install pyspark==3.5.0
+          #pipenv run pip install pyspark==3.5.0
+          pipenv run pip install https://dist.apache.org/repos/dist/dev/spark/v3.5.2-rc3-bin/pyspark-3.5.2.tar.gz
```
Comment (vkorukanti) [inline .github/workflows/spark_test.yaml:64]:
```suggestion
          pipenv run pip install https://dist.apache.org/repos/dist/dev/spark/v3.5.2-rc5-bin/pyspark-3.5.2.tar.gz
```

File: build.sbt [hunk 3]
```diff
@@ -155,6 +155,7 @@ def crossSparkSettings(): Seq[Setting[_]] = getSparkVersion() match {
     targetJvm := "8",
     // For adding staged Spark RC versions, e.g.:
     // resolvers += "Apache Spark 3.5.0 (RC1) Staging" at "https://repository.apache.org/content/repositories/orgapachespark-1444/",
+    resolvers += "Apache Spark 3.5.2 (RC3) Staging" at "https://repository.apache.org/content/repositories/orgapachespark-1459/",
```
Comment (vkorukanti) [inline build.sbt:158]:
```suggestion
    resolvers += "Apache Spark 3.5.2 (RC5) Staging" at "https://repository.apache.org/content/repositories/orgapachespark-1462/",
```

**Review by scottsand-db — APPROVED** [review]

**Review by allisonport-db — CHANGES_REQUESTED** [review]
Can you also update setup.py and anywhere else that is 3.5.0?

**Review by allisonport-db — APPROVED** [review]

**Comment by vkorukanti** [comment]
The Spark master failure is unrelated to this PR.

**Pull request merged**